File: sdkdemo-objc/sdkdemo/ViewController/ScreenBannerViewController.m
//
//  ScreenBannerViewController.m
//  sdkdemo
//
//  Created by Picker Weng on 2024/9/11.
//

#import "ScreenBannerViewController.h"
#import "sdkdemo-Swift.h"

#import <WebKit/WebKit.h>

@import TenMaxMobileAdsSDK;

@interface ScreenBannerViewController () <TenMaxAdSessionDelegate>

@property (nonatomic, strong) WKWebView *webView;
@property (nonatomic, strong) TenMaxAd *screenTopAd;
@property (nonatomic, strong) TenMaxAd *screenTopDuplicatedAd;
@property (nonatomic, strong) TenMaxAd *screenBottomAd;
@property (nonatomic, copy) NSString *screenTopSpaceId;
@property (nonatomic, copy) NSString *screenBottomSpaceId;

@end

@implementation ScreenBannerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.screenTopSpaceId = DemoSettings.screenTop;
    self.screenBottomSpaceId = DemoSettings.screenBottom;
    
    self.webView = [[WKWebView alloc] init];
    self.webView.translatesAutoresizingMaskIntoConstraints = NO;
    
    NSURL *url = [NSURL URLWithString:@"https://www.tenmax.io/tw/"];
    if (url) {
        [self.webView loadRequest:[NSURLRequest requestWithURL:url]];
    }
    
    [self.view addSubview:self.webView];
    
    [NSLayoutConstraint activateConstraints:@[
        [self.webView.leadingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.leadingAnchor],
        [self.webView.trailingAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.trailingAnchor],
        [self.webView.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
        [self.webView.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor],
    ]];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    self.screenTopAd = [TenMaxMobileSDK.shared bannerAdWithSpaceId:self.screenTopSpaceId on:self.view at:TenMaxBannerPositionTop :self :nil];
    self.screenBottomAd = [TenMaxMobileSDK.shared bannerAdWithSpaceId:self.screenBottomSpaceId on:self.view at:TenMaxBannerPositionBottom :self :nil];
    
    [self.screenTopAd show];
    [self.screenBottomAd show];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    
    [self.screenTopAd dispose];
    [self.screenBottomAd dispose];
}

#pragma mark - TenMaxAdSessionDelegate

- (void)adViewableEventSent:(TenMaxAdSession *)session {
    [self makeToastWithText:[NSString stringWithFormat:@"viewable event sent (space id: %@)", session.space.spaceId]];
}

@end

File: sdkdemo/sdkdemo/ScreenBannerViewController.swift
//
//  ScreenBannerViewController.swift
//  sdkdemo
//
//  Created by Picker Weng on 2024/7/28.
//

import UIKit
import WebKit
import TenMaxMobileAdsSDK
import Toast

class ScreenBannerViewController: UIViewController {

    private let screenTopSpaceId = "142f40b224cd48e6"
    private let screenBottomSpaceId = "bc4f97b54f5c425b"
    
    private var webView: WKWebView!
    
    private var screenTopAd: TenMaxAd?
    private var screenBottomAd: TenMaxAd?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        webView = WKWebView()
        webView.translatesAutoresizingMaskIntoConstraints = false
        if let url = URL(string: "https://www.tenmax.io/tw/") {
            webView.load(URLRequest(url: url))
        }
        view.addSubview(webView)
        
        NSLayoutConstraint.activate([
            webView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
            webView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
            webView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
            webView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
        ])
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        screenTopAd = TenMaxMobileSDK.shared().bannerAd(spaceId: screenTopSpaceId, on: view, at: .top, self)
        screenBottomAd = TenMaxMobileSDK.shared().bannerAd(spaceId: screenBottomSpaceId, on: view, at: .bottom, self)
        
        screenTopAd?.show()
        screenBottomAd?.show()
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        
        screenTopAd?.dispose()
        screenBottomAd?.dispose()
    }
}

extension ScreenBannerViewController: TenMaxAdSessionDelegate {
    func adViewableEventSent(_ session: TenMaxAdSession) {
        view.makeToast("viewable event sent (space id: \(session.space.spaceId)")
    }
}
